src/main/java/com/mraof/minestuck/tileentity/redstone/SummonerTileEntity.java

```diff
@@ -97,7 +98,7 @@ public void setSummonedEntity(EntityType<?> entityTypeIn, @Nullable PlayerEntity
 	{
 		this.summonType = entityTypeIn;
 		
-		if(playerEntityIn != null)
+		if(playerEntityIn != null) //used when setting via spawn egg, does not play message when set through gui
 			playerEntityIn.displayClientMessage(new TranslationTextComponent(SUMMON_TYPE_CHANGE, summonType.getRegistryName()), true);
```

Comment:
How about moving this function call to `SummonerBlock.use()`? That'd remove the need for the player parameter and the null check.